fuse/ipns/ipns_unix.go

```diff
@@ -123,12 +123,12 @@ func CreateRoot(n *core.IpfsNode, keys []ci.PrivKey, ipfsroot string) (*Root, er
 			continue
 		}
 
-		if !u.IsValidHash(pointsTo) {
+		if !u.IsValidHash(pointsTo.B58String()) {
```

Comment:
Well, a quick fix would be to make the IsValidHash function just accept a key. Im pretty sure it does a base58 decode inside of it...